lib/measures/tbd/tests/tbd_tests.rb

```diff
@@ -36,187 +36,217 @@ class TBDTest < Minitest::Test
-    # want to use, you don't need it in the hash.
-    args_hash = {"load_tbd_json" => true}
-    # using defaults values from measure.rb for other arguments
+    # Hash of argument values (defaults from measure.rb for other arguments).
+    argh                   = {}
+    argh["load_tbd_json" ] = true
 
-    # populate argument with specified hash value if specified
+    # Populate argument with specified hash value if specified.
     arguments.each do |arg|
       temp_arg_var = arg.clone
-      if args_hash.key?(arg.name)
-        assert(temp_arg_var.setValue(args_hash[arg.name]))
-      end
+      assert(temp_arg_var.setValue(argh[arg.name])) if argh.key?(arg.name)
       argument_map[arg.name] = temp_arg_var
     end
 
-    # run the measure
+    # Run the measure, assert that it did not run correctly.
     Dir.chdir(seed_dir)
     measure.run(model, runner, argument_map)
     result = runner.result
-
-    # show the output
     show_output(result)
+    assert_equal("Fail", result.value.valueName)
 
-    # assert that it ran correctly
-    assert_equal('Fail', result.value.valueName)
-    assert(result.errors.size == 1)
     assert(result.warnings.size == 1)
-    puts result.warnings[0].logMessage
-    log_message = "Can't find 'tbd.json' - simulation halted"
-    assert(result.warnings[0].logMessage == log_message)
+    message = result.warnings[0].logMessage
+    puts message
+    assert(message.include?("Can't find 'tbd.json' - simulation halted"))
+
+    assert(result.errors.size == 1)
+    message = result.errors[0].logMessage
+    puts message
+    assert(message.include?("Halting all TBD processes, "))
+    assert(message.include?("and halting OpenStudio - see 'tbd.out.json'"))
+
+    # Save the model to test output directory.
+    output_path = File.join(seed_dir, "out.osm")
+    model.save(output_path, true)
+  end
+
+  def test_tbd_kiva_massless_error
+    measure = TBDMeasure.new
+
+    # Output directories.
+    seed_dir = File.join(__dir__, "output/tbd_kiva_massless_error/")
+    FileUtils.mkdir_p(seed_dir)
+    seed_path = File.join(seed_dir, "in.osm")
+
+    # Create runner with empty OSW, and example test model.
+    osw = OpenStudio::WorkflowJSON.new
+    osw.setSeedFile(seed_path)
+    runner = OpenStudio::Measure::OSRunner.new(osw)
+    model = OpenStudio::Model::exampleModel
+    model.save(seed_path, true)
+
+    # Copy tdb.json next to seed.
+    origin_pth = File.join(__dir__, "tbd_full_PSI.json")
+    target_pth = File.join(seed_dir, "tbd.json")
+    FileUtils.cp(origin_pth, target_pth)
+
+    # Get measure arguments.
+    arguments = measure.arguments(model)
+    argument_map = OpenStudio::Measure.convertOSArgumentVectorToMap(arguments)
+
+    # Hash of argument values (defaults from measure.rb for other arguments).
+    argh                   = {}
+    argh["gen_kiva_force"] = true
+
+    # POSTULATED USER ERROR : Slab on grade construction holds a massless layer.
```

Comment:
No change brought to the example model. It natively holds a massless material in its slab-on-grade construction, which will trigger an error for KIVA.